app/Model/CoPersonRole.php

```diff
@@ -383,8 +383,9 @@ public function beforeSave($options = array()) {
     }
     
     // If the validity of the role was changed, change the status appropriately
-    
-    if(!empty($this->data[$this->alias]['status'])) {
+    // Skip calculations as part of CO-2195
+    if(!empty($this->data[$this->alias]['status'])
+       && (isset($options['trustStatus']) && !$options['trustStatus'])) {
```

Comment:
I think we want to flip the logic here? `trustStatus` won't be set in most circumstances.

```suggestion
       && (!isset($options['trustStatus']) || $options['trustStatus'])) {
```

Reply:
Done